core/model/modx/processors/element/tv/duplicate.class.php

```diff
@@ -115,20 +115,6 @@ public function duplicateMediaSources() {
         }
     }
 
-    /**
-     * Get the new caption for the duplicate
-     * @return string
-     */
-    public function getNewCaption()
```

Comment:
It might be worth to have a discussion about wether or not removing public methods of a processor constitutes a breaking change*, but until that discussion has been had, I think we should err on the side of caution and not remove any public class methods/variables such as this one in a patch release. 

Could you please restore this method? It can have the new behavior you're doing in `beforeSave`, but the method signature should stay the same. 

---

* According to semver, a change in a "public api" is a breaking change, and a public method of a class does count as a public API... but one could also argue that the "public api" of a processor in MODX is the `$modx->runProcessor($action)` syntax, which isn't affected by this removal in the slightest.

Reply:
@Mark-H What do you offer?
Restore the method which will return the caption?

Reply:
Is that better?